src/app/mod.rs

```diff
@@ -326,10 +327,10 @@ impl Surreal {
 	/// const people = await db.query('SELECT * FROM type::table($table)', { table: 'person' });
 	/// ```
 	pub async fn query(&self, sql: String, bindings: JsValue) -> Result<JsValue, Error> {
-		let bindings: Json = from_value(bindings)?;
-		let mut response = match bindings.is_null() {
-			true => self.db.query(sql).await?,
-			false => self.db.query(sql).bind(bindings).await?,
+		let bindings = json(&from_value::<Json>(bindings)?.to_string()).unwrap_or(Value::None);
+		let mut response = match bindings.is_object() {
+			true => self.db.query(sql).bind(bindings).await?,
+			false => self.db.query(sql).await?,
 		};
```

Comment:
This silently ignores invalid bindings, unlike the previous implementation. The Rust SDK returns an `InvalidBindings` error if the bindings are not an object. That's why the previous implementation forwarded everything as long as there were bindings provided.